public/client.js

```diff
@@ -30,4 +30,10 @@ function removeProduct() {
   userRequest.open('post', '/removeProduct');
   userRequest.setRequestHeader("Content-Type", "application/json;charset=UTF-8")
   userRequest.send(JSON.stringify({'product':document.getElementById("removeProduct").value, 'quantity': document.getElementById("quantity").value}));
-}
\ No newline at end of file
+}
+
+
+function sortbyExpiry() {
+  var x = document.getElementById("expirysort").value;
+  window.location.replace("http://localhost:5000/?expirysort="+x);
```

Comment:
Sorry, I realized I missed something here earlier with this, the problem being that this code will attempt to redirect to localhost:5000, even when it is being run on the live site hosted by Heroku, which we obviously don't want. To fix this for now, please add the following to make the page redirect accordingly:
```
if (window.location.hostname === "localhost") {
    window.location.replace("http://localhost:5000/?expirysort="+x);
  } else {
    window.location.replace("https://whatscookinggoodlooking.herokuapp.com/?expirysort="+x);
  }
```